tests/test_1.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+"""
+There is a kingdom with a king. Kingdom has its own territory with a border. The height of a terrain at the border is
+denoted with a list of integers A. A value from the list A[p] means: when its negative - the terrain level is below the
+sea level, when its positive - the terrain level is above the sea level, when its 0 - the terrain is at the sea level
+height. The king wants to build a castle at each valley and hill along the border of his kingdom. Some part of terrain
+is considered to be a valley when A[p - 1] > A[p]...A[q] < A[q + 1] where q >= p. On the other hand another part of
+terrain is considered to be a hill when A[p - 1] < A[p]...A[q] > A[q + 1]  where q >= p. For beginning and ending of the
+A condition for potential valley or hill is considered to be true. The task is to create a function intended to help the
+king to calculate the number of castles needed. For the list of the same values the function shall return 1 as it is
+considered to be a hill and valley simultaneously.
+"""
+
+import unittest
+
+BORDER_HEIGHT = {
+    'kingdom_1':  [1, 2, 2, 3, 2, 3, 2, 2, 1, 1, 1, 2, 3, 3, 4, 4, 5, 4, 4, 3, 3, 4, 3, 2],
+    'kingdom_2':  [-3, -3, -4, -5, -4, -3, -2, -2, -2, -1, 0, 1, 1, 1, 0, 1, 2, 1, 0, 0, 0, 1, 2, 3, 2, 2],
+    'kingdom_3':  [-5, -5],
+    'kingdom_4':  [1],
+    'kingdom_5':  [1, 2, 1, 2, 1],
+    'kingdom_6':  [1, -2, 1, -2, 1],
+    'kingdom_7':  [1, 2, 1, 1, 1],
+    'kingdom_8':  [1, -2, 1, 1, 1],
+    'kingdom_9':  [1, 1, 1, 2, 1],
+    'kingdom_10': [1, 1, 1, -2, 1],
+    'kingdom_11': [1, 1, 1, 1, 2],
+    'kingdom_12': [2, 1, 1, 1, 1],
+    'kingdom_13': [1, 1, 1, 1, -2],
+    'kingdom_14': [-2, 1, 1, 1, 1],
+    'kingdom_15': [1, 1, 2, 1, 1],
+    'kingdom_16': [1, 1, -2, 1, 1],
+
+}
+EXPECTED_OUTPUT = {
+    'kingdom_1':  9,
+    'kingdom_2':  8,
+    'kingdom_3':  1,
+    'kingdom_4':  1,
+    'kingdom_5':  5,
+    'kingdom_6':  5,
+    'kingdom_7':  3,
+    'kingdom_8':  3,
+    'kingdom_9':  3,
+    'kingdom_10': 3,
+    'kingdom_11': 2,
+    'kingdom_12': 2,
+    'kingdom_13': 2,
+    'kingdom_14': 2,
+    'kingdom_15': 3,
+    'kingdom_16': 3,
+}
+
+
+def castles_number(heights):
+    start_height, current_height = None, None
+    castles = 0
```

Comment:
Why do you need this variables now(here)? Your function can be completed with ```raise Exception``` or ```return 1```